docarray/array/doc_vec/doc_vec.py

```diff
@@ -116,64 +132,113 @@ def __init__(
             # from each document and put them in the corresponding column
             field_type = self.doc_type._get_field_type(field_name)
 
+            is_field_required = self.doc_type.__fields__[field_name].required
+
+            first_doc_is_none = getattr(docs[0], field_name) is None
+
+            def _check_none_field():
+
+                if is_field_required:
+                    if first_doc_is_none:
+                        raise ValueError(
+                            f'Field {field_name} is None for {docs[0]} even though it is required'
+                        )
+
+                if first_doc_is_none:
+                    for i, doc in enumerate(docs):
```

Comment:
I would prefer if this got `docs` passed as an argument, this seems a bit smelly. Plus it "hides" the fact that this fn iterates over all docs